iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/twin/ReportedPropertiesTests.java

```diff
@@ -81,14 +85,14 @@ public void run()
                 {
                     try
                     {
-                        deviceUnderTest.dCDeviceForTwin.createNewReportedProperties(1);
-                        internalClient.sendReportedProperties(deviceUnderTest.dCDeviceForTwin.getReportedProp());
+                        Set<Property> newReportedProperties = deviceUnderTest.dCDeviceForTwin.createNewReportedProperties(1);
+                        internalClient.sendReportedProperties(newReportedProperties);
```

Comment:
It's important to only send this thread's new reported property. If it sends the device's full reported property set, then iothub will receive conflicting sets of properties. 

We would send two twin messages to hub. one with:
```
{
  "key1":"oldValue1",
  "key2":"someUpdatedValue2"
}
```

and another with 

```
{
  "key1":"someUpdatedValue1",
  "key2":"someUpdatedValue2"
}
```

but iothub may process them out of order, and the resulting twin representation on the service side could be either

```
{
  "key1":"someUpdatedValue1",
  "key2":"someUpdatedValue2"
}
```

or the incorrect representation

```
{
  "key1":"oldValue1",
  "key2":"someUpdatedValue2"
}
```

Reply:
The test would fail in the latter case because it was expecting a final state of 
```
{
  "key1":"someUpdatedValue1",
  "key2":"someUpdatedValue2"
}
```

Reply:
So the fix here is to just send one property at a time, as it gets updated.

```
{
  "key1":"someUpdatedValue1"
}
```

and

```
{
  "key2":"someUpdatedValue2"
}
```

so that iothub's representation of this twin will always be 

```
{
  "key1":"someUpdatedValue1",
  "key2":"someUpdatedValue2"
}
```